x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -243,9 +248,24 @@ protected LogicalPlan rule(Aggregate aggregate) {
             LogicalPlan plan = aggregate;
             if (changed) {
                 var source = aggregate.source();
-                plan = new Aggregate(aggregate.source(), aggregate.child(), aggregate.groupings(), newAggs);
+                if (newAggs.isEmpty() == false) {
+                    plan = new Aggregate(source, aggregate.child(), aggregate.groupings(), newAggs);
+                } else {
+                    // All aggs actually have been surrogates for (foldable) expressions, e.g.
+                    //
+                    // ...
+                    // | STATS a = AVG([1,2]), m = min([1,2,3])
+                    //
+                    // The output needs to be 1 row containing the results of the folding.
+                    // To achieve this, we replace the aggregation with a single row and let the subsequent Eval produce the values, so
+                    // the plan becomes essentially
+                    //
+                    // ROW placeholder = null
+                    // | EVAL a = MV_AVG([1,2]), m = MV_MIN([1,2,3])
```

Comment:
What Andrei said. Better to have this comment inside SurrogateExpression for future guidance - here it should be just one or two lines, explaining what's going on.